src/braket_container.py

```diff
@@ -149,14 +150,39 @@ def kick_off_customer_script(entry_point : str) -> multiprocessing.Process:
         str_module, _, str_method = entry_point.partition(":")
         customer_module = importlib.import_module(str_module)
         customer_method = getattr(customer_module, str_method)
-        customer_code_process = multiprocessing.Process(target=customer_method)
+
+        process_kwargs = {"target": customer_method}
+
+        function_args = try_bind_hyperparameters_to_customer_method(customer_method)
+        if function_args is not None:
+            process_kwargs["kwargs"] = function_args
+
+        customer_code_process = multiprocessing.Process(**process_kwargs)
         customer_code_process.start()
     except Exception as e:
         log_failure_and_exit(f"Unable to run job at entry point {entry_point}\nException: {e}")
     return customer_code_process
 
 
-def join_customer_script(customer_code_process : multiprocessing.Process):
+def try_bind_hyperparameters_to_customer_method(customer_method: Callable):
+    hp_file = os.getenv("AMZN_BRAKET_HP_FILE")
+    with open(hp_file) as f:
```

Comment:
Should we check if the file isn't actually specified? Or what was the reason that you did not put this part inside the try statement?

Reply:
This is a stable part of the jobs API, whether or not hyperparameters are given (just confirmed with local/aws jobs). The try/except block exists very specifically to catch a TypeError on the `bind` method when hyperparameters do not line up with function parameters.

Reply:
Cleaned up the code to make this more clear

Reply:
I'm worried about SM changing it behind the scenes where if there are no HP then the file will not be given/created. In other parts of the code we protect against this, should we do the same here?

Reply:
To me, it seems that removing this file would be a breaking change. For example, a customer could have code:
```
# hp defaults
hps = {
    "hp_1": 1,
    "hp_2": 2,
    ...
}
provided_hps = load_hps_from_file()
hps.update(provided_hps)
```
and expect this to work whether or not hyperparameters are provided for that job run.

For that reason, it feels like we can assume that we will keep this contract and not get rid of that hp file. I believe that is something we own as a service and not created by sagemaker. There are a couple gaps in my knowledge of how exactly this file gets created, so happy to discuss further

Reply:
Yes, but I think that's a contrived example. Most customers that don't use hyperparameters wouldn't have code to handle them.

Reply:
Either way, doesn't hurt to have the check. Added